### PR TITLE
explorer-show-recent-package-component

### PR DESCRIPTION
**Pull request by Jibz1**
Screenshot 2022-11-12 at 7 38 49 PM

### CODE REVIEW

**Comment by mystie711** [comment]
The Figma designs have the section lower down on the page,
https://www.figma.com/file/kSD0qXCIDZA4KQVNepA5hR/02-Explorer?node-id=29%3A13465

File: apps/explorer/src/components/recent-packages-card/RecentPackagesCard.tsx
Comment (Jordan-Mysten) [inline apps/explorer/src/components/recent-packages-card/RecentPackagesCard.tsx:3]: Thanks for using this!

File: apps/explorer/src/components/recent-packages-card/RecentPackagesCard.tsx
Comment (Jordan-Mysten) [inline apps/explorer/src/components/recent-packages-card/RecentPackagesCard.tsx:45]: This seems like something that could be defined outside of this function

File: apps/explorer/src/components/recent-packages-card/RecentPackagesCard.tsx
Comment (Jordan-Mysten) [inline apps/explorer/src/components/recent-packages-card/RecentPackagesCard.tsx:26]: Can hoist this outside of the component itself

File: apps/explorer/src/components/recent-packages-card/RecentPackagesCard.tsx
Comment (Jordan-Mysten) [inline apps/explorer/src/components/recent-packages-card/RecentPackagesCard.tsx:53]: I'd be open to a new `InternalLink` (probably need a better name) that took a `type` like address, object, etc and linked there, and also managed consistent truncation. Basically `LongText` but rebuilt on new UI components.

You don't need to do this now, we can create a backlog item for it.

File: apps/explorer/src/components/recent-packages-card/RecentPackagesCard.tsx
Comment (Jordan-Mysten) [inline apps/explorer/src/components/recent-packages-card/RecentPackagesCard.tsx:99]: Don't do this in the query result, do this with the data response afterwards, for example:

```
const table = useMemo(() => data ? transformTable(data) : null, [data]);
```

File: apps/explorer/src/components/tx-time/TxTimeType.tsx
Comment (Jordan-Mysten) [inline apps/explorer/src/components/tx-time/TxTimeType.tsx:10]: We could also make this a base UI component and port over the time utilities.

File: apps/explorer/src/components/recent-packages-card/RecentPackagesCard.tsx
Comment (Jibz1) [reply apps/explorer/src/components/recent-packages-card/RecentPackagesCard.tsx:53]: Agreed, we can just build that out later.

File: apps/explorer/src/components/tx-time/TxTimeType.tsx
Comment (Jibz1) [reply apps/explorer/src/components/tx-time/TxTimeType.tsx:10]: We can combine this component with the DateCard component on the other PR

**Review by mystie711 — APPROVED** [review]

File: apps/explorer/src/components/recent-packages-card/RecentPackagesCard.tsx
Comment (Jordan-Mysten) [inline apps/explorer/src/components/recent-packages-card/RecentPackagesCard.tsx:103]: Do we also want auto-refreshing here? cc @mystie711 curious what your thoughts are.

File: apps/explorer/src/components/recent-packages-card/RecentPackagesCard.tsx
Comment (Jordan-Mysten) [inline apps/explorer/src/components/recent-packages-card/RecentPackagesCard.tsx:47]: I'd just guard before to make this return easier to understand:

```typescript
if (!('publish' in event)) return {}; // or null or whatever works
```

File: apps/explorer/src/components/recent-packages-card/RecentPackagesCard.tsx
Comment (Jordan-Mysten) [inline apps/explorer/src/components/recent-packages-card/RecentPackagesCard.tsx:45]: Just destructure in the function arguments itself

**Review by Jordan-Mysten — APPROVED** [review]
Approving to unblock, but please address the comments before merging

**Pull request merged**
